pallets/messages/src/rpc/src/lib.rs

```diff
@@ -100,7 +99,7 @@ where
 		'loops: for block_number in from..to {
 			let list: Vec<MessageResponse> = api
 				.get_messages_by_schema_and_block(
-					&at,
+					self.client.info().best_hash,
```

Comment:
I think this is a function call and not memoized, so we'd need to keep the at so it doesn't get called n times.

Reply:
Nice catch, I didn't catch the loop part, focused on the BlockId removal.